File: module/common/src/main/java/com/example/common/network/base/NetBaseBean.java
package com.example.common.network.base;

import androidx.annotation.NonNull;

import com.example.network.bean.NetBean;
import com.example.network.utils.GsonUtil;

import java.io.Serializable;

/**
 * 理论上所有接口都应继承该数据bean
 */
public class NetBaseBean extends NetBean implements Serializable {

    private String returnCode;
    private String returnMsg;

    public String getReturnCode() {
        return returnCode == null ? "" : returnCode;
    }

    public void setReturnCode(String returnCode) {
        this.returnCode = returnCode;
    }

    public String getReturnMsg() {
        return returnMsg == null ? "" : returnMsg;
    }

    public void setReturnMsg(String returnMsg) {
        this.returnMsg = returnMsg;
    }

    public boolean isSuccess() {
        return getReturnCode().equals(INetRequestCode.SUCCESS);
    }

    public boolean isToLogin() {
        return getReturnCode().equals(INetRequestCode.TO_LOGIN);
    }

    @NonNull
    @Override
    public String toString() {
        return GsonUtil.GsonToString(this);
    }
}

File: module/common/src/main/java/com/example/common/network/base/NetBaseBean.kt
package com.example.common.network.base

import com.example.network.bean.NetBean
import com.example.network.utils.GsonUtil.GsonToString
import java.io.Serializable

/**
 * 理论上所有接口都应继承该数据bean
 */
open class NetBaseBean : NetBean(), Serializable {
    var returnCode: String? = null
        get() = if (field == null) "" else field
    var returnMsg: String? = null
        get() = if (field == null) "" else field

    val isSuccess: Boolean
        get() = returnCode == INetRequestCode.SUCCESS

    val isToLogin: Boolean
        get() = returnCode == INetRequestCode.TO_LOGIN

    override fun toString(): String {
        return GsonToString(this)
    }
}